wurst/network/SyncSimple.wurst

```diff
@@ -0,0 +1,170 @@
+package SyncSimple
+
+import MapBounds
+import UnitIndexer
+import ErrorHandling
+import BitSet
+import LinkedListModule
+
+/********************************************************************************
+	SyncSimple library v1.0 by MoriMori.
+
+	Based on TriggerHappy's SyncInteger library.
+
+	Rationale:
+		This library can be used to 'synchronize' players after some async network
+		action has been performed by one or more players.
+
+		It abuses the fact that all network events in WC3 are fired sequentially,
+		meaning that they arrrive for other players in the order that they were
+		sent in.
+		It also abuses the fact that the EVENT_PLAYER_UNIT_SELECTED fires
+		synchronously for all players at the same time, allowing us to know for
+		certain when other players have acknowledged our unit selection, as well
+		as all network events that have been fired before it.
+		This is due to the fact that the WC3 game protocol is built on top of TCP,
+		which is a sequential stream protocol - meaning that sent data is always
+		received in the same order that it was sent, allowing us to do what we do.
+
+		By calling the .sync() method, we queue a network action (specifically,
+		a unit selection event) that will only be delivered after all previously 
+		queued network actions have also been delivered.
+		The primary use of this library is in conjunction with gamecache's Sync
+		natives, because they also fire sequential network events. When we call
+		.sync() after a series of Sync natives, we ensure that the .onSynced()
+		callback will only be called after all players have received the data.
+
+		This way, we can send local data from one player to the rest, such as
+		camera position, data from files, and so on.
+
+		There may be other usages related to async network events as well.
+
+	Usage:
+		1. Create a new SimpleSynchronizer using:
+			let synchronizer = new SimpleSynchronizer()
+
+		2. Send some network events on one (or more) player using the Sync natives (or something else)
+			if sender == localPlayer
+				// value1 and value2 are some local values only known to localPlayer
+				SyncInteger(gc, mkey, vkey1, value1)
+				SyncInteger(gc, mkey, vkey1, value1)
+		
+		3. Call .sync() on all players after calling the Sync natives
```

Comment:
"on all players?" is this the line I need, or do I need more

Reply:
"on all players" in this case means that each player has to call it in order to proceed with the sync. If one or more player doesn't call it, e.g. in a localPlayer clause, then it will not sync.

Reply:
Maybe "Call `.sync()` for all game clients, after having called the Sync natives."